src/main/java/com/commercetools/project/sync/SyncerFactory.java

```diff
@@ -55,122 +61,172 @@ public static SyncerFactory of(
+              "Unknown argument \"%s\" supplied to \"-%s\" or \"--%s\" option! %s",
+              null,
+              SYNC_MODULE_OPTION_SHORT,
+              SYNC_MODULE_OPTION_LONG,
+              SYNC_MODULE_OPTION_DESCRIPTION);
+      throw new IllegalArgumentException(errorMessage);
+    }
+    if (syncOptionValues.length == 1 && SYNC_MODULE_OPTION_ALL.equals(syncOptionValues[0])) {
+      return Arrays.asList(SyncModuleOption.values());
+    } else {
+      return Arrays.stream(syncOptionValues)
+          .map(
+              syncOption -> {
+                if (isBlank(syncOption)) {
+                  final String errorMessage =
+                      format(
+                          "Blank argument supplied to \"-%s\" or \"--%s\" option! %s",
+                          SYNC_MODULE_OPTION_SHORT,
+                          SYNC_MODULE_OPTION_LONG,
+                          SYNC_MODULE_OPTION_DESCRIPTION);
 
-      return exceptionallyCompletedFuture(new IllegalArgumentException(errorMessage));
+                  throw new IllegalArgumentException(errorMessage);
+                }
+
+                final String trimmedValue = syncOption.trim();
+                try {
+                  return SyncModuleOption.getSyncModuleOptionBySyncOptionValue(trimmedValue);
+                } catch (IllegalArgumentException e) {
+                  final String errorMessage =
+                      format(
+                          "Unknown argument \"%s\" supplied to \"-%s\" or \"--%s\" option! %s",
+                          trimmedValue,
+                          SYNC_MODULE_OPTION_SHORT,
+                          SYNC_MODULE_OPTION_LONG,
+                          SYNC_MODULE_OPTION_DESCRIPTION);
+                  throw new IllegalArgumentException(errorMessage);
+                }
+              })
+          .collect(Collectors.toList());
     }
+  }
 
-    Syncer<
-            ? extends ResourceView,
-            ?,
-            ? extends BaseSyncStatistics,
-            ? extends BaseSyncOptions<?, ?>,
-            ? extends QueryDsl<?, ?>,
-            ? extends BaseSync<?, ?, ?>>
-        syncer;
+  private static Collection<List<SyncModuleOption>> groupSyncModuleOptions(
+      List<SyncModuleOption> syncModuleOptions) {
+    Map<SyncModuleOption, Integer> syncModuleOptionWithReferencesCount =
+        syncModuleOptions
+            .stream()
+            .collect(
+                Collectors.toMap(
+                    key -> key,
+                    option ->
+                        (option.getReferences().isEmpty())
+                            ? 0
+                            : countReferences(option.getReferences(), syncModuleOptions)));
 
-    try {
-      syncer = buildSyncer(syncOptionValue, runnerNameOptionValue, isSyncProjectSyncCustomObjects);
-    } catch (IllegalArgumentException exception) {
+    Map<Integer, List<SyncModuleOption>> groupedSyncModuleOptions = new TreeMap<>();
```

Comment:
maybe you can explain why order matters here

Reply:
https://github.com/commercetools/commercetools-project-sync/pull/219/commits/39701b0756b605aa70a49b931d101e18bacc033f